scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -45,10 +45,13 @@ usage() {
 		  force     Manually force leaderlog calculation and overwrite even if already done, exits after leaderlog is calculated
 		validate    Continously monitor and confirm that the blocks made actually was accepted and adopted by chain (deployed as service)
 		  all       One-time re-validation of all blocks in blocklog db
-		  epoch     One-time re-validation of blocks in blocklog db for the specified epoch 
+		  epoch     One-time re-validation of blocks in blocklog db for the specified epoch
+		epochdata   Manually re-calculate leaderlog to load stakepool history into epochdata table of blocklog db. Needs completion of cncli.sh sync and cncli.sh validate processes.
+		  all       One-time re-calculation of all epochs (avg execution duration: 1hr / 50 epochs)
+		  epoch     One-time re-calculation for the specified epoch
```

Comment:
I know I'm late to the review party but have a few comments. Please don't take this as a request for any changes, it's only intended to open a discussion between reviewers/maintainers.

I also have some results from testing I will post separately.
----
The **epochdata**  command seems very tightly related to the **leaderlog** command.  It seems like it could be easy to overlook for the average user while they review **leaderlog** options. 

From solely a user perspective (not considering the current code or complexity to change things) it seems like  **all** and **epoch** subcommands would make more sense being an extension of the **leaderlog** command since it appears to be specific to only adjusting the table where leaderlog may not have been calculated, or was calculated incorrectly.

Reply:
The initial idea of mine was to have an option to load full pool history once, epochdata table will then get informed at every leaderlog check, i don't disagree the idea of make it part of leaderlog function, personally I would not run it as service for first automatic load after the cncli db full sync and of historical load of blocklog table, since it could take several hours to load all the information, 2hr/140epochs approx.

For sure it would be easier for users, epochdata gets loaded by default, then you have the option to do a manual reload, or a single reload (quite useless feature imo, but may be can be useful in some cases).

On last point of single epoch choice I'm going to add: print available epochs from blocklog, otherwise some user could input an epoch for which there were no assigned slots, and get message: slots not found for epoch...

Reply:
it could be a service that run only once and dependent from cncli-sync and cncli-validate or cncli-leaderlog, the one that fill blocklog table.

The only question I have on this, it is fine to run automatic process that could take hours on a BP that has several blocks every epochs? The process runs "cncli leaderlog" at every iteration, I noticed from htop that is quite loady, but may be not to the point to affect negatively the BP node.

Reply:
Hello Team, I added "show all available epochs" on wrong epoch number input. So instead of print epochs at every "cncli.sh epochdata num" that looks ugly to me, is printed on wrong (epoch not in blocklog) input and should be fine to get the list once (or at every wrong input) to know pool's epochs.

![image](https://github.com/user-attachments/assets/af38583c-9801-4cf2-8128-253e14e297ad)

Reply:
> The initial idea of mine was to have an option to load full pool history once, epochdata table will then get informed at every leaderlog check, i don't disagree the idea of make it part of leaderlog function, personally I would not run it as service for first automatic load after the cncli db full sync and of historical load of blocklog table, since it could take several hours to load all the information, 2hr/140epochs approx.

When first contemplating the feature, I wasn't considering the **init**  command. It makes sense to not combine with the **leaderlog** command given **init** exists and the requirement of the **validate** command has been run.

Reply:
Just to double-check - Is this conversation OK to resolve?

Reply:
Hello @rdlrt 

I think yes